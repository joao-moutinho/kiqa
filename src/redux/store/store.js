import { combineReducers, createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import cartReducer from "../reducer";
import logReducer from "../reducer/loginred";

const rootReducers = combineReducers({ cart: cartReducer, loggedUser: logReducer });

export const store = createStore(
  rootReducers,
  composeWithDevTools(applyMiddleware())
);
