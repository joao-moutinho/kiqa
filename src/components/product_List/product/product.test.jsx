/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
/* eslint-disable testing-library/prefer-screen-queries */
import { fireEvent, render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { BrowserRouter, Router } from 'react-router-dom';
import {Product} from '.';
import {MemoryRouter} from 'react-router-dom'
import { Provider } from 'react-redux';
import { store } from '../../../redux/store/store';

const product = {
    id: 1,
    price : 5,
    description: "",
    brand: "",
    name: "",
    isActive: true,
    image: "  ",
    categoryName: "",
    productType: "",
    colors: [
        {
            "hexValue": "#B28378",
            "colourName": "BFF PENCIL" 
        }
    ]
}

describe("Product", () => {

    test("render", () => {
        const {container} = render(<BrowserRouter><Provider store={store}><Product product={product} /></Provider> </BrowserRouter>); 
        expect(container).toMatchSnapshot();
    })

    test("Should showModal when press Add to Cart button",()=> {
        const {container} = render(<BrowserRouter><Provider store={store}><Product product={product} /></Provider> </BrowserRouter>);
        fireEvent.click(screen.getByText(/Add to Cart/i));
        expect(container.getElementsByClassName("modal-buy-box")).toMatchSnapshot();
        expect(container).toMatchSnapshot();
    })

    test("Not Should show Modal",()=> {
        const {container} = render(<BrowserRouter><Provider store={store}><Product product={product} /></Provider> </BrowserRouter>);
        fireEvent.click(screen.getByText(/Add to Cart/i));
        fireEvent.click(screen.getByTestId("modal-buy"));
        expect(container).toMatchSnapshot();
    })

    test("Should navigate to /products/id when press image div", () =>{
        const history = createMemoryHistory();
        const {container} = render(<BrowserRouter><Provider store={store}><Product product={product} /></Provider> </BrowserRouter>);
        fireEvent.click(container.querySelector(".product-image"));
    })

})