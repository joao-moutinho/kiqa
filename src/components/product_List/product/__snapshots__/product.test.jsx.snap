// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Not Should show Modal 1`] = `
<div>
  <div
    class="product"
  >
    <div
      class="product-image"
    >
      <img
        height="150px"
        src="  "
        width="150px"
      />
    </div>
    <div
      class="product-name"
    />
    <div
      class="product-price"
    >
      5
       €
    </div>
    <div
      class="buy-button"
    >
      <div
        class="primary-button"
      >
        <button>
          Add to Cart
        </button>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`Product Should showModal when press Add to Cart button 1`] = `
HTMLCollection [
  <div
    class="modal-buy-box"
  >
    <div
      class="modal-buy-box-image"
    >
      <img
        alt="product_photo"
        src="  "
        width="95%"
      />
    </div>
    <div
      class="modal-buy-box-rigth"
    >
      <div
        class="modal-buy-box-rigth-info"
      >
        <div
          class="modal-buy-box-rigth-info-name"
        />
        <div
          class="modal-buy-box-rigth-info-price"
        >
          5
           €
        </div>
      </div>
      <div
        class="modal-buy-box-rigth-dropdownColors"
      >
        <div
          class="modal-buy-box-rigth-dropdownColors-color"
          data-testid="color"
          style="background-color: rgb(178, 131, 120); border: 2px solid black;"
        />
      </div>
      <div
        class="modal-buy-box-rigth-buy"
      >
        <div
          class="modal-buy-box-rigth-buy-quantity"
        >
          <div
            class="quantity"
          >
            <button
              class="quantity-button"
            >
               
              -
               
            </button>
            <div
              class="quantity-output"
            >
              1
            </div>
            <button
              class="quantity-button"
            >
               
              +
               
            </button>
          </div>
          <div
            class="primary-button"
          >
            <button>
              Add to Cart
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Product Should showModal when press Add to Cart button 2`] = `
<div>
  <div
    class="product"
  >
    <div
      class="product-image"
    >
      <img
        height="150px"
        src="  "
        width="150px"
      />
    </div>
    <div
      class="product-name"
    />
    <div
      class="product-price"
    >
      5
       €
    </div>
    <div
      class="buy-button"
    >
      <div
        class="primary-button"
      >
        <button>
          Add to Cart
        </button>
      </div>
    </div>
    <div
      class="modal-buy"
      data-testid="modal-buy"
    >
      <div
        class="modal-buy-box"
      >
        <div
          class="modal-buy-box-image"
        >
          <img
            alt="product_photo"
            src="  "
            width="95%"
          />
        </div>
        <div
          class="modal-buy-box-rigth"
        >
          <div
            class="modal-buy-box-rigth-info"
          >
            <div
              class="modal-buy-box-rigth-info-name"
            />
            <div
              class="modal-buy-box-rigth-info-price"
            >
              5
               €
            </div>
          </div>
          <div
            class="modal-buy-box-rigth-dropdownColors"
          >
            <div
              class="modal-buy-box-rigth-dropdownColors-color"
              data-testid="color"
              style="background-color: rgb(178, 131, 120); border: 2px solid black;"
            />
          </div>
          <div
            class="modal-buy-box-rigth-buy"
          >
            <div
              class="modal-buy-box-rigth-buy-quantity"
            >
              <div
                class="quantity"
              >
                <button
                  class="quantity-button"
                >
                   
                  -
                   
                </button>
                <div
                  class="quantity-output"
                >
                  1
                </div>
                <button
                  class="quantity-button"
                >
                   
                  +
                   
                </button>
              </div>
              <div
                class="primary-button"
              >
                <button>
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`Product render 1`] = `
<div>
  <div
    class="product"
  >
    <div
      class="product-image"
    >
      <img
        height="150px"
        src="  "
        width="150px"
      />
    </div>
    <div
      class="product-name"
    />
    <div
      class="product-price"
    >
      5
       €
    </div>
    <div
      class="buy-button"
    >
      <div
        class="primary-button"
      >
        <button>
          Add to Cart
        </button>
      </div>
    </div>
  </div>
   
</div>
`;
